lib/mayaUsd/utils/utilFileSystem.cpp

```diff
@@ -750,3 +750,56 @@ std::string UsdMayaUtilFileSystem::pathFindExtension(std::string& filePath)
     ghc::filesystem::path ext = p.extension();
     return ext.string();
 }
+
+UsdMayaUtilFileSystem::FileBackup::FileBackup(const std::string& filename)
```

Comment:
Automatic backup/restore of a file. Taken from code that was in the layer editor and generalized to be reusable.